homeassistant/components/sql/config_flow.py

```diff
@@ -82,8 +84,10 @@
 
 def validate_sql_select(value: str) -> str | None:
     """Validate that value is a SQL SELECT query."""
-    if not value.lstrip().lower().startswith("select"):
-        raise ValueError("Incorrect Query")
+    query_type = sqlparse.parse(value)[0].get_type().upper()
```

Comment:
Included this check with a specific validation error

Reply:
Looks good.
Could we also maybe cleanup the value string to start with ensuring it doesn't start with `;` which some have the habit to add?
If someone does the length will be more than 1 and it will be annoying for them to fix and we can skip to get issues just for a simple thing like that.

Reply:
I included this, but this cleaning is only being done during query validation to prevent the error. Do you think it is better for this treatment to be included so that the query is stored clean?

Reply:
Yes, can we store it cleaned we should do so.

Reply:
Included in https://github.com/home-assistant/core/pull/105921/commits/27c7738ffe7d17456e5c939abdc296c0221c5d92.

For the YAML config, I don't see what to do other than validate and present the error.